sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -177,6 +178,36 @@ def test_fail_example(log_collector):
             raise ValueError('Did not stop executing script after error')
 
 
+class _FakeApp(object):
+    def __init__(self):
+        for key in ('info', 'warn', 'error', 'critical'):
+            setattr(self, key, lambda msg, **kwargs: logging.info(msg))
+
+    def status_iterator(self, iterable, *args, **kwargs):
+        for it in iterable:
+            yield it
+
+
+def test_gen_dir_rst():
+    """Test gen_dir_rst."""
+    _app = sphinx_compatibility._app
```

Comment:
This is because PR #250 uses a sphinx status iterator to reduce the amount of text sent to stdout when building the gallery in `gen_dir_rst`. Sphinx changed the API for that iterator, thus there is also a compatibility layer provided by `sphinx_compatibility` 
This function must use the `pytest.fixture already` in place instead of repeating what is already implemented in there:
```python
def test_gen_dir_rst(fakesphinxapp):
```
and in sphinx_gallery/tests/conftest.py change it to:

```python
class FakeSphinxApp:
    def __init__(self):
        self.calls = collections.defaultdict(list)

    def status_iterator(self, *args, **kwargs):
        self.calls['status_iterator'].append(Params(args, kwargs))
        for it in args[0]:
            yield it

```